### ISSUE TITLE
The code of contour snap

### ISSUE THREAD

**Issue by ZhiQue930716** [kmaninis/OSVOS-PyTorch#33]
Hi,
I read your paper, but I can not find the part of code about contour snap which expressed in paper.  Especially，the Fast Bilateral Solver(FBS) and the CNN trained to detect object contour.
Could you show me that. Thanks.

**Comment by scaelles** [comment]
The boundary snapping gives a boost of around 2.5% on the IoU metric as shown in the paper. Therefore, it is a refinement of the appearance model (the one that this code reproduces) and not the core of the algorithm. The boundary snapping is implemented in Matlab and implementing it in python would require quite a lot of time so we are not planning to do it. If you would like to use it, you can try the [caffe](https://github.com/kmaninis/OSVOS-caffe) version.

In the paper we also tried another refinement using the Fast Bilateral Solver which also gives a nice (but smaller than boundary snapping) improvement and there is code available in [python](https://github.com/poolio/bilateral_solver). Thus, I would recommend you to give it a try.
